File: cotsSource/org.apache.qpid/java/client/src/old_test/java/org/apache/qpid/headers/Listener.java
/*
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 * 
 *   http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 *
 */
package org.apache.qpid.headers;

import org.apache.qpid.client.AMQConnection;
import org.apache.qpid.client.AMQSession;
import org.apache.qpid.jms.Session;
//import org.apache.qpid.testutil.Config;

import javax.jms.MessageListener;
import javax.jms.Message;
import javax.jms.Destination;
import javax.jms.MessageProducer;
import javax.jms.JMSException;

public class Listener //implements MessageListener
{
/*    private final AMQConnection _connection;
    private final MessageProducer _controller;
    private final AMQSession _session;
    private final MessageFactory _factory;
    private int count;
    private long start;

    Listener(AMQConnection connection, Destination exchange) throws Exception
    {
        _connection = connection;
        _session = (AMQSession) connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
        _factory = new MessageFactory(_session, 0, 19);

        //register for events
        _factory.createConsumer(exchange).setMessageListener(this);
        _connection.start();

        _controller = _session.createProducer(exchange);
    }

    private void shutdown()
    {
        try
        {
            _session.close();
            _connection.stop();
            _connection.close();
        }
        catch(Exception e)
        {
            e.printStackTrace(System.out);
        }
    }

    private void report()
    {
        try
        {
            String msg = getReport();
            _controller.send(_factory.createReportResponseMessage(msg));
            System.out.println("Sent report: " + msg);
        }
        catch(Exception e)
        {
            e.printStackTrace(System.out);
        }
    }

    private String getReport() throws JMSException
    {
        long time = (System.currentTimeMillis() - start);
        return "Received " + count + " in " + time + "ms";
    }

    public void onMessage(Message message)
    {
        if(count == 0) start = System.currentTimeMillis();

        if(_factory.isShutdown(message))
        {
            shutdown();
        }
        else if(_factory.isReport(message))
        {
            //send a report:
            report();
        }
        else if (++count % 100 == 0)
        {
            System.out.println("Received " + count + " messages.");
        }
    }

    public static void main(String[] argv) throws Exception
    {
        Config config = new Config();
        config.setType(Config.HEADERS);
        config.setName("test_headers_exchange");
        config.setOptions(argv);
        new Listener((AMQConnection) config.getConnection(), config.getDestination());
    }*/
}

File: cotsSource/org.apache.qpid/python/examples/direct/listener.py
#!/usr/bin/env python
#
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
# 
#   http://www.apache.org/licenses/LICENSE-2.0
# 
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.
#
"""
 listener.py

 This AMQP client reads messages from a message
 queue named "message_queue". It is implemented
 as a message listener.
"""

# Common includes

import qpid
import sys
import os
from qpid.util import connect
from qpid.connection import Connection
from qpid.datatypes import Message, RangedSet, uuid4
from qpid.queue import Empty

# Includes specific to this example

from time         import sleep


#----- Message Receive Handler -----------------------------
class Receiver:
  def __init__ (self):
    self.finalReceived = False

  def isFinal (self):
    return self.finalReceived
    
  def Handler (self, message):
    content = message.body
    session.message_accept(RangedSet(message.id)) 
    print content
    if content == "That's all, folks!":
      self.finalReceived = True

#----- Initialization --------------------------------------

#  Set parameters for login

host="127.0.0.1"
port=5672
user="guest"
password="guest"

# If an alternate host or port has been specified, use that instead
# (this is used in our unit tests)
if len(sys.argv) > 1 :
  host=sys.argv[1]
if len(sys.argv) > 2 :
  port=int(sys.argv[2])

#  Create a connection.
socket = connect(host, port)
connection = Connection (sock=socket, username=user, password=password)
connection.start()
session = connection.session(str(uuid4()))

#----- Read from queue --------------------------------------------

# Now let's create a local client queue and tell it to read
# incoming messages.

# The local_queue_name identifies the client-side queue.

local_queue_name = "local_queue"
queue = session.incoming(local_queue_name)

# Call message_subscribe() to tell the broker to deliver messages
# from the AMQP queue to this local client queue. The broker will
# start delivering messages as soon as message_subscribe() is called.

session.message_subscribe(queue="message_queue", destination=local_queue_name)
queue.start()

receiver = Receiver()
queue.listen (receiver.Handler)

while not receiver.isFinal() :
  sleep (1)


#----- Cleanup ------------------------------------------------

# Clean up before exiting so there are no open threads.
#

session.close(timeout=10)
